test/suite/test_txn20.py

```diff
@@ -0,0 +1,121 @@
+        # not protected from dirty reads so we'll see the update above even
+        # though its respective transaction has not been committed.
+        s = self.conn.open_session()
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction('isolation=read-uncommitted')
+        self.assertEqual(cursor['key: aaa'], 'value: bbb')
+
+        # Commit the update now. We should still see it from our
+        # 'read-uncommitted' session.
+        self.session.commit_transaction()
+        self.assertEqual(cursor['key: aaa'], 'value: bbb')
+
+        # Cleanup.
+        self.session.close()
+        s.rollback_transaction()
+        s.close()
+
+    def test_read_committed(self):
+        self.session.create(self.uri, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(self.uri, None)
+        cursor['key: aaa'] = 'value: aaa'
+
+        # Make an update and don't commit it just yet.
+        self.session.begin_transaction()
+        cursor['key: aaa'] = 'value: bbb'
+
+        # Now begin a new transaction with the 'read-committed' isolation level.
+        # We shouldn't see the update above since it hasn't been committed yet.
+        s = self.conn.open_session()
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction('isolation=read-committed')
+        self.assertEqual(cursor['key: aaa'], 'value: aaa')
+
+        # Commit the update now. Unlike the 'snapshot' isolation level, we're
+        # not protected from non-repeatable reads so we'll see an update that
+        # wasn't visible earlier in our previous read.
+        self.session.commit_transaction()
+        self.assertEqual(cursor['key: aaa'], 'value: bbb')
+
+        # Cleanup.
+        self.session.close()
+        s.rollback_transaction()
+        s.close()
+
+    def test_read_snapshot(self):
+        self.session.create(self.uri, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(self.uri, None)
+        cursor['key: aaa'] = 'value: aaa'
+
+        # Make an update and don't commit it just yet.
+        self.session.begin_transaction()
+        cursor['key: aaa'] = 'value: bbb'
+
+        # Now begin a new transaction with the 'snapshot' isolation level.
+        # We should never see the update above regardless of what happens
+        # from here on since it wasn't committed at the time of the snapshot.
+        s = self.conn.open_session()
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction('isolation=snapshot')
+        self.assertEqual(cursor['key: aaa'], 'value: aaa')
+
+        # Commit the update now. We still shouldn't see the value since it wasn't
+        # part of the initial snapshot.
+        self.session.commit_transaction()
+        self.assertEqual(cursor['key: aaa'], 'value: aaa')
+
+        # Cleanup.
+        self.session.close()
+        s.rollback_transaction()
+        s.close()
+
+if __name__ == '__main__':
+    wttest.run()
```

Comment:
Between different test cases, it seems one or two lines are different. In that case we can use make_scenario's utility to reduce the test code. If not clear how to do it, we can discuss.

Reply:
How does this help? It looks like `make_scenarios` generates a series of operations whereas I have a specific operation that I want to test.

I can put some of the code in some `setUp` and `tearDown` functions if that's helpful.

Reply:
cc6c6fe